vector/src/main/java/im/vector/app/features/location/LocationTracker.kt

```diff
@@ -38,18 +40,17 @@ class LocationTracker @Inject constructor(
         fun onLocationProviderIsNotAvailable()
     }
 
-    private var callback: Callback? = null
+    private var callbacks = mutableListOf<Callback>()
 
     private var hasGpsProviderLiveLocation = false
 
     @RequiresPermission(anyOf = [Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION])
-    fun start(callback: Callback?) {
+    fun start() {
```

Comment:
now that we're calling `start` from multiple places, do we need to guard the...

```kotlin
                    locationManager.requestLocationUpdates(
                            provider,
                            MIN_TIME_TO_UPDATE_LOCATION_MILLIS,
                            MIN_DISTANCE_TO_UPDATE_LOCATION_METERS,
                            this
                    )
```  

to avoid registering duplicate updates?

Reply:
It is protected internally and doesn't create multiple requests.

Reply:
good to know, thanks!